src/util/child-component.js

```diff
@@ -103,3 +89,43 @@ export function createChildComponent (definition={}) {
 export function rejectNullElements(children) {
 	return _.reject(React.Children.toArray(children), (node) => (React.isValidElement(node) && node.type.isRenderNull));
 }
+
+export function filterElementsByType(children, elementTypes) {
+	return _.filter(React.Children.toArray(children), (node) => (React.isValidElement(node) && _.includes(elementTypes, node.type)));
+}
+
+export function getChildComponentPropsArray(props, childComponentType) {
+	const propName = _.get(childComponentType, 'childComponentDefinition.propName');
+	if (propName) {
+		const currentProps = _.chain(props)
+			.get(propName) // grab the prop we care about, e.g. "Child"
+			.thru(x => [x]) // wrap in an array
+			.flatten()
+			.filter(x => x) // remove falsey values
+			.value();
+		const childComponentPropsArray = _.map(currentProps, (childProp) => {
+			if (_.isPlainObject(childProp)) {
+				return childProp;
+			} else {
+				return {
+					children: childProp
+				};
+			}
+		});
+		return childComponentPropsArray;
+	}
+	return [];
+}
+
+export function findAllChildComponents(props, childComponentTypes) {
+	return _.reduce(childComponentTypes, (acc, childComponentType) => {
+		const childComponentProps = getChildComponentPropsArray(props, childComponentType);
+		return acc.concat(_.map(childComponentProps, (childComponentProp) => ({
+			type: childComponentType,
+			props: childComponentProp
+		})));
+	}, []).concat(_.map(filterElementsByType(props.children, childComponentTypes), (element) => ({
+		type: element.type,
+		props: element.props
+	})));
+}
```

Comment:
actually, now that i think about it, the map isn't really needed...i like the fact that it might be an actual element because you'll know if that component came from children since `React.isValidElement` would return true for it, so u kind of get extra data for free